### PR TITLE
Add support for custom ports to the Ruby debugger.

### PR DESCRIPTION
**Pull request by ukblewis**
Make sure these boxes are checked before submitting your PR -- thanks in advance!

- [] Build pass!
- [] Follow VS Code [Coding Guidelines](https://github.com/Microsoft/vscode/wiki/Coding-Guidelines)

### CODE REVIEW

**Comment by ukblewis** [comment]
Can I have some help with getting the tests to pass @rebornix ?

**Comment by rebornix** [comment]
@ukblewis I'll take a look

**Comment by ukblewis** [comment]
@rebornix Thank you - any progress so far?

File: src/ruby.ts [hunk 1]
```diff
@@ -218,7 +222,7 @@ export class RubyProcess extends EventEmitter {
 
             this.debugprocess.stderr.on('data', (data: Buffer) => {
                 if (/^Fast Debugger/.test(data.toString())) {
-                    this.debugSocketClient.connect(1234);
+                    this.debugSocketClient.connect(args.debuggerPort);
```
Comment (rebornix) [inline src/ruby.ts]: `args.debuggerPort` might be `undefined` so you may want to use `args.debuggerPort | 1234`, that's why the test cases fail.

File: .travis.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ before_install:
  - nvm install v5.10.0
  - nvm use v5.10.0
 before_script:
-  - npm install -g typescript
+  - npm install -g typescript@1.8.2
```
Comment (rebornix) [inline .travis.yml]: I already fix this in master, do you mind reverting this line of change.

**Comment by rebornix** [comment]
LGTM, just two tiny comments and they should be real easy fix.

**Comment by rebornix** [comment]
@ukblewis thanks, it's really awesome!

**Pull request merged**